instrumentation/activemq-client-5.8.0/src/main/java/com/nr/agent/instrumentation/activemqclient580/ActiveMQUtil.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ *  * Copyright 2024 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package com.nr.agent.instrumentation.activemqclient580;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.agent.bridge.messaging.BrokerInstance;
+
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ActiveMQUtil {
+
+    private static final ActiveMQUtil INSTANCE = ActiveMQUtil.create();
+
+    public static ActiveMQUtil get() {
+        return INSTANCE;
+    }
+
+    private final Pattern addressPattern = Pattern.compile("^\\w+://(.+)/.+:(\\d+)");
+
+    public BrokerInstance parseHostAndPort(String address) {
+        return doParseHostAndPort(address);
+    }
+
+    private final Function<String, BrokerInstance> CACHE = AgentBridge.collectionFactory.memorize(this::doParseHostAndPort, 32);
```

Comment:
It looks like this CACHE is never actually used... ?  I'm guessing it was meant to be called above in parseHostAndPort.  And then the doParseHostAndPort was meant to be private?